src/main/java/com/hubspot/ringleader/watcher/PersistentWatcher.java

```diff
@@ -79,15 +79,15 @@ public void start() {
     if (started.compareAndSet(false, true)) {
       fetchInExecutor();
 
-      executor.schedule(() -> {
+      executor.scheduleAtFixedRate(() -> {
         int versionBeforeFetch = lastVersion.get();
         fetch();
 
         if (lastVersion.get() != versionBeforeFetch) {
           LOG.error("Detected a change that didn't raise an event; replacing curator");
           parent.replaceCurator();
         }
-      }, 10, TimeUnit.MINUTES);
+      }, 1, 1, TimeUnit.MINUTES);
```

Comment:
Is there a race in here with the background update and this fetch? Probably still not that likely and I guess in the worst case we just replace the curator client?

Reply:
I think you're right that there is, but if it takes >1 minute to do the initial fetch we probably want to replace the curator client anyway